clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/view/KubernetesProviderUtils.groovy

```diff
@@ -65,4 +68,30 @@ class KubernetesProviderUtils {
     }
     return instanceMap
   }
+
+  static KubernetesServerGroup serverGroupFromCacheData(ObjectMapper objectMapper, CacheData cacheData, Set<KubernetesInstance> instances) {
+    ReplicationController replicationController
+    ReplicaSet replicaSet
+    if (cacheData.attributes.replicationController) {
+      replicationController = objectMapper.convertValue(cacheData.attributes.replicationController, ReplicationController)
+    } else if (cacheData.attributes.replicaSet) {
+      replicaSet = objectMapper.convertValue(cacheData.attributes.replicaSet, ReplicaSet)
+    } else {
+      throw new IllegalStateException("Expected either a ReplicaSet or ReplicationController")
```

Comment:
Just a nit, but the order of the resource types in the error message don't match the order of the comparisons. Same thing down below.